processor/src/main/java/com/squareup/sqlbrite/SqlBriteProcessor.java

```diff
@@ -0,0 +1,152 @@
+    filer = processingEnv.getFiler();
+    messager = processingEnv.getMessager();
+  }
+
+  @Override
+  public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+    for (Element annotated : roundEnv.getElementsAnnotatedWith(Table.Builder.class)) {
+      processType(annotated);
+    }
+    return false;
+  }
+
+  private void processType(Element element) {
+    if (element.getKind() != ElementKind.INTERFACE) {
+      messager.printMessage(Diagnostic.Kind.ERROR, "Can only annotate " + Table.Builder.class +
+          " on interfaces.", element);
+      return;
+    }
+
+    TypeElement typeElement = (TypeElement) element;
+    if (typeElement.getModifiers().contains(Modifier.PRIVATE)) {
+      messager.printMessage(Diagnostic.Kind.ERROR, "Can't annotate private interfaces with " +
+          Table.Builder.class, element);
+      return;
+    }
+
+    String builderName = classNameOf(typeElement).replaceAll("\\.", "_") + "Impl";
+    TypeSpec.Builder builder = TypeSpec.classBuilder(builderName)
+        .addSuperinterface(TypeName.get(typeElement.asType()));
+    FieldSpec field = FieldSpec.builder(CONTENT_VALUES_CLASS, "values")
+        .initializer("new $T()", CONTENT_VALUES_CLASS).build();
+    builder.addField(field);
+
+    String packageName = packageNameOf(typeElement);
+    ClassName builderClassName = ClassName.get(packageName, builderName);
+    for (ExecutableElement method : ElementFilter.methodsIn(typeElement.getEnclosedElements())) {
+      final Column column = method.getAnnotation(Column.class);
+      if (column != null) {
+        String columnName = column.value();
+        List<? extends VariableElement> params = method.getParameters();
+        if (params.size() == 1) {
+          VariableElement var = params.get(0);
+          MethodSpec builderMethod = MethodSpec.methodBuilder(method.getSimpleName().toString())
+              .addModifiers(Modifier.PUBLIC)
+              .returns(builderClassName)
+              .addAnnotation(Override.class)
+              .addParameter(TypeName.get(var.asType()), "param")
+              .addStatement("values.put($S, param)", columnName)
+              .addStatement("return this")
+              .build();
+          builder.addMethod(builderMethod);
+        }
+      }
+    }
+
+    MethodSpec buildMethod = MethodSpec.methodBuilder("build")
+        .addModifiers(Modifier.PUBLIC)
+        .addAnnotation(Override.class)
+        .returns(CONTENT_VALUES_CLASS)
+        .addStatement("return values")
+        .build();
+    builder.addMethod(buildMethod);
+
+    try {
+      JavaFile javaFile = JavaFile.builder(packageName, builder.build()).build();
+      javaFile.writeTo(filer);
+    } catch (Exception e) {
+      messager.printMessage(Diagnostic.Kind.ERROR, "Unable to write file", element);
+    }
+  }
+
+  /**
+   * Returns the name of the given type, including any enclosing types but not the package.
+   *
+   * Taken from AutoValue
```

Comment:
Are these methods in [auto/common][1]? If so, just add a dependency on it.

 [1]: https://github.com/google/auto/tree/master/common